library/alloc/src/collections/vec_deque.rs

```diff
@@ -32,12 +32,8 @@ mod tests;
 
 const INITIAL_CAPACITY: usize = 7; // 2^3 - 1
 const MINIMUM_CAPACITY: usize = 1; // 2 - 1
-#[cfg(target_pointer_width = "16")]
-const MAXIMUM_ZST_CAPACITY: usize = 1 << (16 - 1); // Largest possible power of two
-#[cfg(target_pointer_width = "32")]
-const MAXIMUM_ZST_CAPACITY: usize = 1 << (32 - 1); // Largest possible power of two
-#[cfg(target_pointer_width = "64")]
-const MAXIMUM_ZST_CAPACITY: usize = 1 << (64 - 1); // Largest possible power of two
+
+const MAXIMUM_ZST_CAPACITY: usize = 1 << (core::mem::size_of::<usize>() * 8 - 1); // Largest possible power of two
```

Comment:
Why isn't this just `usize::MAX`?

Reply:
That's a good point, and I now realize an issue that was present in the original code, it appears the -1 is misplaced. I'll fix both, thank you.

Reply:
I was wrong -- as the comment says it wants a power of two, not `MAX`.

Reply:
It's alright, confused me though.